graph/loaders/assets.js

```diff
@@ -54,58 +37,121 @@ const getAssetsByQuery = async (context, query) => {
   };
 };
 
-/**
- * This endpoint find or creates an asset at the given url when it is loaded.
- * @param   {Object} context   the context of the request
- * @param   {String} asset_url the url passed in from the query
- * @returns {Promise}          resolves to the asset
- */
-const findOrCreateAssetByURL = async (context, asset_url) => {
-
-  // Verify that the asset_url is parsable.
-  let parsed_asset_url = url.parse(asset_url);
-  if (!parsed_asset_url.protocol) {
-    throw errors.ErrInvalidAssetURL;
+const findOrCreateAssetByURL = async (ctx, url) => {
+
+  // Pull our connectors out of the context.
+  const {
+    loaders: {
+      Assets,
+      Settings,
+    },
+    connectors: {
+      models: {
+        Asset,
+      },
+      services: {
+        DomainList,
+        Scraper,
+      },
+      errors: {
+        ErrInvalidAssetURL,
+      },
+    },
+  } = ctx;
+
+  // Try to validate that the url is valid. If the URL constructor throws an
+  // error, throw our internal ErrInvalidAssetURL instead. This will validate
+  // that the url contains a valid scheme.
+  try {
+    new URL(url);
+  } catch (err) {
+    throw ErrInvalidAssetURL;
   }
 
-  let asset = await AssetsService.findOrCreateByUrl(asset_url);
```

Comment:
Whats the reason behind not implementing the changes in `AssetsService.findOrCreateByUrl`. Is it still required?